test/framework/options.py

```diff
@@ -747,6 +747,27 @@ def test_search(self):
         self.mock_stdout(False)
         self.assertTrue(re.search('GCC-4.9.2', txt))
 
+        # test using a search pattern that includes special characters like '+', '(', or ')' (should not crash)
+        # cfr. https://github.com/easybuilders/easybuild-framework/issues/2966
+        # characters like ^, . or * are not touched, since these can be used as regex characters in queries
+        for opt in ['--search', '-S', '--search-short']:
+            for pattern in ['netCDF-C++', 'foo|bar', '^foo', 'foo.*bar']:
+                args = [opt, pattern, '--robot', test_easyconfigs_dir]
+                self.mock_stdout(True)
+                self.eb_main(args, raise_error=True, verbose=True, testing=False)
+                stdout = self.get_stdout()
+                self.mock_stdout(False)
+                # there shouldn't be any hits for any of these queries, so empty output...
```

Comment:
Eh, the netCDF-C++ should find something and thus not have empty output, or am i not understanding what you're doing here?

Reply:
There's no `netCDF-C++` easyconfig in the test easyconfigs that are included in the framework repo (we don't use the easyconfigs repo for running the tests)

Reply:
Ok, I thought it was working against the normal repo...

Reply:
@akesandgren a5e47fb68 should help...